### DISCUSSION TITLE
Failed to look up dependency Service in kubernetes

### DISCUSSION THREAD

**Discussion by DawidJanczak** [renovatebot/renovate#18222] [Request Help]
Hello,

We started getting some warnings around certain types of kubernetes resources, e.g.
```
Failed to look up dependency Service (Service)(packageFile="deployment/k8s/base/service.yaml", dependency="Service")
```

This makes sense to me, but I wonder why certain resources produce this warnings (services, config maps, kustomizations, secrets) where others that I think should do not (pod disruption budgets, ingresses).

I wonder if it would be helpful to mention in the documentation that it's only deployments and bare pods (anything else?) that setting renovate with kubernetes makes sense for.

**Comment by gberche-orange** [comment]
I'm observing the same symptom with many K8S dependencies now detected (see below). 

Looking at the renovate release notes, it seems to correlate with https://github.com/renovatebot/renovate/issues/5972 implemented in https://github.com/renovatebot/renovate/pull/17420 and released in https://github.com/renovatebot/renovate/releases/tag/32.221.0 on 2022-10-06, see https://docs.renovatebot.com/modules/datasource/#kubernetes-api-datasource

```
INFO: Dependency extraction complete
[...]
"kubernetes": {
  "fileCount": 1237,
  "depCount": 1292
},
```

On our project, this has the side effect of breaking the rendering of the dependency dashboard issue, that gets truncated. See screenshots below

![image](https://user-images.githubusercontent.com/4748380/194826357-299fa71d-12e7-4d9d-86ce-157684e1c8b6.png)
[...]
![image](https://user-images.githubusercontent.com/4748380/194826485-cc14a2a0-7bde-4335-8dc8-3eb10ad713ba.png)

/CC @JamieMagee @rarkins

**Reply by gberche-orange** [reply]
Update: sorry, I had misread the job traces, this rather seems to be related to the k8s manager rather than the k8s-api datasource. I'll continue trying to read release notes and recent changes to understand if a renovate changed triggered the symptom we are observing

**Reply by viceice** [reply]
@JamieMagee I think it's because the new api datasource doesn't know all possible api versions. they can be user extended by CRD's. so we should probably return an empty releases list for unknown api's

**Reply by viceice** [reply]
- https://github.com/renovatebot/renovate/pull/18230

**Reply by gberche-orange** [reply]
@viceice @rarkins I did not yet find a distinct issue for tracking `too long content breaks rendering of the Dependency Dashboard gh issue` already exists. I measured in this case the issue body was truncated at 59,999 chars and wondered who truncated the content.

According to https://github.com/orgs/community/discussions/27190#discussioncomment-3726017 the current github limit for body is larger (one report of  262,144 chars being properly displayed). So I guess the limit is on renovate's side.

That'd be worth improving, so that the useful content in the dependency dashboards still get displayed, or devising some kind of priority among source of content in the dashboard when hitting the github limit.

**Reply by HonkingGoose** [reply]
> @viceice @rarkins I did not yet find a distinct issue for tracking too long content breaks rendering of the Dependency Dashboard gh issue already exists.

We had an old issue about the truncating of extra content in the Dependency Dashboard. We closed that issue because we did not have a reproduction:

- #16373

By the way, I commented on your new issue with a list of some more related issues / pull request. You and the maintainers should probably review that list, to see what ideas we had before:

- https://github.com/renovatebot/renovate/issues/18244#issuecomment-1273855574
